pkg/baremetal/redfish/redfish.go

```diff
@@ -0,0 +1,90 @@
+package redfish
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/litmuschaos/litmus-go/pkg/log"
+)
+
+// State helps get the power state of the node
+type State struct {
+	PowerState string
+}
+
+// NodestatusCheck will check and return the status of the node.
+func GetNodeStatus(IP, user, password string) string {
+	URL := fmt.Sprintf("https://%v/redfish/v1/Systems/System.Embedded.1/", IP)
+	auth := user + ":" + password
+	encodedAuth := base64.StdEncoding.EncodeToString([]byte(auth))
+	data := map[string]string{}
+	json_data, _ := json.Marshal(data)
+	req, err := http.NewRequest("GET", URL, bytes.NewBuffer(json_data))
+	if err != nil {
+		msg := fmt.Sprintf("Error creating http request: %v", err)
+		log.Error(msg)
+		return "Unknown"
```

Comment:
Can we return the error here?  Please check this for all other places.
```suggestion
		return  errors.Errorf("fail to get the node status, err: %v", msg)
```